db/fetcher/indexer_iterators.go

```diff
@@ -501,38 +431,176 @@ func createIndexIterator(
 			}
 			valArr = append(valArr, valueBytes)
 		}
-		if op == opIn {
-			var iter filterValueIndexIterator
-			if isUnique {
-				iter = &eqSingleIndexIterator{
-					indexKey: indexDataStoreKey,
-					execInfo: execInfo,
-				}
-			} else {
-				iter = &eqPrefixIndexIterator{
-					indexKey: indexDataStoreKey,
-					execInfo: execInfo,
-				}
+		return newNinIndexCmp(valArr, op == opIn), nil
+	case opLike, opNlike:
+		return newLikeIndexCmp(filterVal.(string), op == opLike), nil
+	case opAny:
+		return &anyMatcher{}, nil
+	}
+
+	return nil, errors.New("invalid index filter condition")
+}
+
+func createValueMatchers(conditions []fieldFilterCond) ([]valueMatcher, error) {
+	matchers := make([]valueMatcher, 0, len(conditions))
+	for i := range conditions {
+		m, err := createValueMatcher(conditions[i].op, conditions[i].val)
+		if err != nil {
+			return nil, err
+		}
+		matchers = append(matchers, m)
+	}
+	return matchers, nil
+}
+
+type fieldFilterCond struct {
+	op  string
+	val any
+}
+
+func (f *IndexFetcher) determineFieldFilterConditions() []fieldFilterCond {
+	result := make([]fieldFilterCond, 0, len(f.indexedFields))
+	for i := range f.indexedFields {
+		fieldInd := f.mapping.FirstIndexOfName(f.indexedFields[i].Name)
+		found := false
+		for filterKey, indexFilterCond := range f.indexFilter.Conditions {
+			propKey, ok := filterKey.(*mapper.PropertyIndex)
+			if !ok {
+				continue
```

Comment:
todo: I can't spot any good reason for this ever happening, and codecov says it is untested.  Should this not be an error?  Or perhaps can we cast here without checking type?  If neither of those can you please add a comment documenting how this can happen, and why it is okay if it does.

EDIT: When looking at the next `fieldInd != propKey.Index` if, I think this is just you cycling through trying to find the correct property, however the cast raises eyebrows and it's intent is not that clear - if this the reason for this code, can I suggest that instead of the current you do the below, as it makes it clearer that the cast is just a requirement for finding the correct prop:

```
if !ok || fieldInd != propKey.Index {
    // this is not the property we are looking for (reword :))
    continue
}
```

Reply:
merged ifs and added a comment

Reply:
cheers, looks good